agent/job_runner.go

```diff
@@ -299,11 +308,22 @@ func (r *JobRunner) Run() error {
 		jobMetrics.Count(`jobs.failed`, 1)
 	}
 
+	if signalReason == "" {
+		if r.stopped {
+			// The agent is being gracefully stopped, and we signalled the job to end. Often due
+			// to pending host shutdown or EC2 spot instance termination
+			signalReason = `agent_stop`
+		} else if r.cancelled {
+			// The job was signalled because it was cancelled via the buildkite web UI
+			signalReason = `cancel`
+		}
+	}
```

Comment:
Perhaps this logic could go where we set `exitStatus` from the process?

https://github.com/buildkite/agent/blob/be13cdb015da221f61ddec772846a292f1d70e0c/agent/job_runner.go#L260-L267

Reply:
nice idea. I've put it inside the conditional that sets `signal` - it feels weird to leave a possibility that `signalReason` is set but `signal` is blank...

Reply:
I guess `signalReason == ""` is synonymous for "unknown", i.e. the job might have been sent a signal by something outside the agent's control?